### ISSUE TITLE
Users are shown but grant Permissions throws DirectoryServicesCOMException: An operations error occurred

### ISSUE THREAD

**Issue by BeniFreitag** [Yvand/LDAPCP#130]
People Picker shows user correctly with their emailaddress but when I grant permissions to any user an error is shown. ULS logs these messages (screenshot below):

[] Unexpected error occurred while getting domain names information for LDAP connection LDAP://mydomain.local/DC=mydomain,DC=local (DirectoryServicesCOMException): System.DirectoryServices.DirectoryServicesCOMException: An operations error occurred.  , Callstack:   
 at System.DirectoryServices.DirectoryEntry.Bind(Boolean throwIfFail)    
 at System.DirectoryServices.DirectoryEntry.Bind()    
 at System.DirectoryServices.DirectoryEntry.get_AdsObject()    
 at System.DirectoryServices.PropertyCollection.Contains(String propertyName)    
 at ldapcp.OperationContext.GetDomainInformation(DirectoryEntry directory, String& domaindistinguishedName, String& domainName, String& domainFQDN)

[LDAPCP] Unexpected error occurred while connecting to LDAP server LDAP://mydomain.local/DC=mydomain,DC=local: System.DirectoryServices.DirectoryServicesCOMException: An operations error occurred.  , Callstack:   
 at System.DirectoryServices.DirectoryEntry.Bind(Boolean throwIfFail)    
 at System.DirectoryServices.DirectoryEntry.Bind()    
 at System.DirectoryServices.DirectoryEntry.get_AdsObject()    
 at System.DirectoryServices.DirectorySearcher.FindAll(Boolean findMoreThanOne)    
 at ldapcp.LDAPCP.<>c__DisplayClass43_0.<QueryLDAPServers>b__1(LDAPConnection ldapConnection)

[LDAPCP] Validation failed: found 0 entities instead of 1 for incoming claim with value 'benjamin.test@...' and type 'http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress'

It only occurs when I use "Connect to SharePoint AD domain" (default) but works when I use "Connect to a LDAP server" with the same LDAP-Path LDAP://mydomain.local/DC=mydomain,DC=local and the same Process account.

LDAPCP Version: 14.1.20191007.981
SharePoint Server 2019, May 2020 CU

![image](https://user-images.githubusercontent.com/6669093/95960621-3e21e980-0e04-11eb-947f-bb7fddb4165a.png)
![image](https://user-images.githubusercontent.com/6669093/95959826-4a597700-0e03-11eb-9530-0123d38c2883.png)
![image](https://user-images.githubusercontent.com/6669093/95959838-4d546780-0e03-11eb-9b78-625782479894.png)

**Comment by Yvand** [comment]
* Does it repro with all accounts? Including groups?
* What version of LDAPCP are you using?
* Did you define custom LDAP connections?
* Can you switch LDAPCP logs to verbose to get the exact LDAP query and test it yourself?
  * To switch to verbose: `"LDAPCP:*"| Set-SPLogLevel -TraceSeverity Verbose`
  * To replay query: https://ldapcp.com/Troubleshoot-LDAPCP.html

**Comment by BeniFreitag** [comment]
* It occours with all user accounts.
* LDAPCP Version: 14.1.20191007.981 (latest)
* When I use a custom LDAP connection it works. It only fails when I use the standard LDAP connection.

Verbose Logging shows this additional message:
[] Unexpected error occurred while getting domain names information for LDAP connection LDAP://mydomain.local/DC=mydomain,DC=local (DirectoryServicesCOMException): System.DirectoryServices.DirectoryServicesCOMException: An operations error occurred. , Callstack:
at System.DirectoryServices.DirectoryEntry.Bind(Boolean throwIfFail)
at System.DirectoryServices.DirectoryEntry.Bind()
at System.DirectoryServices.DirectoryEntry.get_AdsObject()
at System.DirectoryServices.PropertyCollection.Contains(String propertyName)
at ldapcp.OperationContext.GetDomainInformation(DirectoryEntry directory, String& domaindistinguishedName, String& domainName, String& domainFQDN)

Replay query with PowerShell works without any issue, it shows me the desired users. Also reading the Domain Properties works in PowerShell, as the Farm Account:
````powershell
Add-Type -AssemblyName System.DirectoryServices.AccountManagement
$ldapPath = "LDAP://mydomain.local/DC=mydomain,DC=local"
$de = New-Object System.DirectoryServices.DirectoryEntry ($ldapPath)
$de.Properties["distinguishedName"].Value.ToString()
````
Output is: `DC=mydomain,DC=local`

**Comment by Yvand** [comment]
I'm a little bit out of suggestion here, I can't see what difference could be relevant between the custom LDAP connection and the default one...
What options do you use when you create the custom LDAP connection? Can you try using options "Secure" and "Signing"?

**Comment by BeniFreitag** [comment]
For the custom LDAP connection I used just "Secure". Otherwise I recieve "A more secure authentication method is required for this server. ". "Secure" and "Signing" also works.

We have multiple SharePoint Servers with LDAPCP and SharePoint 2013/2016, all in the same Active Directory and didn't run into this issue before.

Anyway, since I figured out the workaround to just add the default-connection once again as a custom LDAP connection, we could also close this issue.
